Upstream-Status: Inappropriate
Signed-off-by: Tomas Frydrych <tomas@sleepfive.com>
Updated from patch by Phil Blundell <philb at gnu.org>

Use makekeys from -native package rather than trying to compile it here, and 
fix up configure logic to find keysymdef.h in the sysroot.

Index: git/Makefile.am
===================================================================
--- git.orig/Makefile.am	2012-10-10 10:27:51.935448294 +0100
+++ git/Makefile.am	2012-10-10 10:27:52.227447093 +0100
@@ -1,5 +1,5 @@
 # Order: makekeys before src
-SUBDIRS = include makekeys src test
+SUBDIRS = include src test
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xkbcommon.pc
Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2012-10-10 10:27:51.939448276 +0100
+++ git/configure.ac	2012-10-10 10:27:52.227447093 +0100
@@ -74,7 +74,8 @@
 
 # Obtain the path to the X Window System Core Protocol
 AC_MSG_CHECKING([for X11 includedir])
-includex11dir="`$PKG_CONFIG --variable=includex11dir xproto`"
+m4_pattern_allow([PKG_CONFIG_SYSROOT_DIR])
+includex11dir="$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=includex11dir xproto`"
 AC_MSG_RESULT([$includex11dir])
 
 AC_MSG_CHECKING([keysym definitions])
Index: git/src/Makefile.am
===================================================================
--- git.orig/src/Makefile.am	2012-10-10 10:27:51.939448276 +0100
+++ git/src/Makefile.am	2012-10-10 10:27:52.227447093 +0100
@@ -7,6 +7,7 @@
 libxkbcommon_la_LIBADD = xkbcomp/libxkbcomp.la
 libxkbcommon_la_LDFLAGS = -no-undefined
 libxkbcommon_la_SOURCES = \
+	$(BUILT_SOURCES) \
 	XKBcommonint.h \
 	alloc.c \
 	atom.c \
@@ -25,5 +26,5 @@
 BUILT_SOURCES = ks_tables.h
 CLEANFILES = $(BUILT_SOURCES)
 
-ks_tables.h: $(KEYSYMDEFS) $(top_builddir)/makekeys/makekeys$(EXEEXT)
-	$(top_builddir)/makekeys/makekeys $(KEYSYMDEFS) > $@
+ks_tables.h: $(KEYSYMDEFS)
+	 libxkbcommon-makekeys$(EXEEXT) $(KEYSYMDEFS) > $@
